pages/first-principles.markdown

```diff
@@ -85,6 +93,8 @@ When you start a new codebase, the most important thing is not getting Feature A
 
 All codebases look great when they're new and ~2,000 LOC. It takes discipline to have a mature codebase that is 50,000 LOC, 100,000 LOC, still look good, and be a productive, pleasant experience to development in.
 
+:speech_balloon: Hmm, I would formulate this a bit differently and phrase it as "Design and code like the code base will live for another five years." I think that it's an antipattern to reduce quality investments just because the system is nearing its projected EOL. In my experience, it's worth keeping that investment level high because (if nothing else) you preserve the *option* of extending the lifetime of the code base, which may happen regardless.
```

Comment:
Yes, agreed, I didn't mean "your codebase will live for exactly 5 years, so you can stop worrying at year 4.5.". :-) I'll think about a clarification.